packages/databases-collections-list/src/grid-header.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useContext, useMemo } from 'react';
+import { css, cx, spacing, Breadcrumbs } from '@mongodb-js/compass-components';
+import type { BreadcrumbItem } from '@mongodb-js/compass-components';
+import toNS from 'mongodb-ns';
+import { useOpenWorkspace } from '@mongodb-js/compass-workspaces/provider';
+
+const gridControlItemStyles = css({
+  flex: 'none',
+});
+
+const pushRightStyles = css({
+  marginLeft: 'auto',
+});
+
+const gridControlContainerStyles = css({
+  display: 'flex',
+  gap: spacing[2],
+});
+
+// We use this context to pass components that are aware of outer state of the
+// v-list to the list header. This is needed so that we can define this outer
+// component outside of the list component scope and avoid constant re-mounts
+// when component constructor is re-created. We do this so that controls can be
+// part of the list header and scroll up when the list is scrolled
+export const ControlsContext = React.createContext<{
+  createControls: React.ReactElement | null;
+  refreshControls: React.ReactElement | null;
+  viewTypeControls: React.ReactElement | null;
+  sortControls: React.ReactElement | null;
+}>({
+  createControls: null,
+  refreshControls: null,
+  viewTypeControls: null,
+  sortControls: null,
+});
+
+const GridControls = () => {
+  const { createControls, refreshControls, viewTypeControls, sortControls } =
+    useContext(ControlsContext);
+
+  return (
+    <div className={gridControlContainerStyles}>
+      {createControls && (
+        <div className={gridControlItemStyles} data-testid="create-controls">
+          {createControls}
+        </div>
+      )}
+      {refreshControls && (
+        <div className={gridControlItemStyles} data-testid="refresh-controls">
+          {refreshControls}
+        </div>
+      )}
+      <div className={cx(gridControlItemStyles, pushRightStyles)}>
+        {viewTypeControls}
+      </div>
+      <div className={gridControlItemStyles}>{sortControls}</div>
+    </div>
+  );
+};
+
+const containerStyles = css({
+  display: 'flex',
+  flexDirection: 'column',
+  flex: 1,
+  gap: spacing[3],
+});
+
+const breadcrumbStyles = css({
+  display: 'flex',
+});
+
+export const GridHeader = ({ namespace }: { namespace?: string }) => {
+  const { openDatabasesWorkspace, openCollectionsWorkspace } =
+    useOpenWorkspace();
+
+  const breadcrumbItems = useMemo(() => {
+    return [
+      {
+        // TODO (COMPASS-7684): add connection name
```

Comment:
Not sure if it's worth discussing with design, but while we wait for the multiconnection work to handle this, maybe we can hide it for now?

Reply:
I'll check it with the design and update this (and the todo ticket) accordingly.

Reply:
I removed it from this PR in this [b95a406](https://github.com/mongodb-js/compass/pull/5501/commits/b95a406c9f2e46f059dc50148683ecc81748406e). I added this for reference in the ticket. So now we are only showing breadcrumb on the collection-header, without `Cluster`